impeller/entity/contents/content_context.h

```diff
@@ -729,10 +729,42 @@ class ContentContext {
     return render_target_cache_;
   }
 
+  std::shared_ptr<Pipeline<PipelineDescriptor>> GetCachedRuntimeEffectPipeline(
```

Comment:
The immediate thought that came to mind was, "Aren't all runtime stage entrypoints just called 'main'?" I suppose we mangle taking into account the name of the stage. But that still means that different stages with the same name will collide. I am not sure if that issue is purely hypothetical but it will be hard to debug if we run into it.

I suppose we could hash the name and the structure of the inputs before in the compiler and include that in the fbs. Or just generate a UUID.

Reply:
How the uniqueness of the entrypoint is guaranteed today would be good to add to a docstring.

Reply:
I'll try to document this. We build up the entrypoint name from the file name in the compiler.

Reply:
I was going to try to plumb through the asset name for this but realized we basically already have that in the entrypoint name.

Reply:
A single type-4 UUID would be fine honestly. But I suspect this works fine for now. Filing a followup works.

Reply:
https://github.com/flutter/flutter/issues/141304